packages/mattermost-redux/src/actions/teams.ts

```diff
@@ -76,14 +76,39 @@ export function getMyTeams(): ActionFunc {
     });
 }
 
-export function getMyTeamUnreads(collapsedThreads: boolean): ActionFunc {
-    return bindClientFunc({
-        clientFunc: Client4.getMyTeamUnreads,
-        onSuccess: TeamTypes.RECEIVED_MY_TEAM_UNREADS,
-        params: [
-            collapsedThreads,
-        ],
-    });
+// The argument skipCurrentTeam is a (not ideal) workaround for CRT mention counts. Unread mentions are stored in the reducer per
+// team but we do not track unread mentions for DMs/GMs independently. This results in a bit of funky logic and edge case bugs
+// that need workarounds like this. In the future we should fix the root cause with better APIs and redux state.
+export function getMyTeamUnreads(collapsedThreads: boolean, skipCurrentTeam = false): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let unreads;
+        try {
+            unreads = await Client4.getMyTeamUnreads(collapsedThreads);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        if (skipCurrentTeam) {
+            const currentTeamId = getCurrentTeamId(getState());
+            if (currentTeamId) {
+                const index = unreads.findIndex((member) => member.team_id === currentTeamId);
+                if (index >= 0) {
+                    unreads.splice(index, 1);
+                }
+            }
```

Comment:
I'm wondering if this should instead happen on the reducer instead of here, as the reducer is the one in charge of merging the data. 0/5 though.

Reply:
Yeah, I struggled with where to put but I landed on keeping it here since the comment I wrote has all the context for it and it means it's less spaghetti'ed throughout different files

Reply:
yeah, as I said 0/5.